src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -796,12 +791,12 @@ private void TrackNullableStateForAssignment(BoundExpression value, TypeSymbolWi
                         // When that issue is fixed, Nullable<T> should be handled there instead.
                         if (valueSlot > 0)
                         {
-                            InheritNullableStateOfTrackableType(targetSlot, valueSlot, slotWatermark);
+                            InheritNullableStateOfTrackableType(targetSlot, valueSlot, stopAtSlot: targetSlot);
```

Comment:
>stopAtSlot: targetSlot [](start = 87, length = 22)

I'm wondering whether we could remove the `stopAtSlot` parameter from `InheritNullableStateOfTrackableType` and `InheritNullableStateOfTrackableStruct` (and maybe even `InheritNullableStateOfMember`) and just always use the `targetSlot` parameter for that purpose in those methods.

Reply:
We want to stop at the original target slot even the recursive case (when calling those same methods to copy members).

---
In reply to: [258218756](https://github.com/dotnet/roslyn/pull/33455#discussion_r258218756) [](ancestors = 258218756)